compiler/sem/semexprs.nim

```diff
@@ -2193,26 +2211,26 @@ proc semAsgn(c: PContext, n: PNode; mode=asgnNormal): PNode =
           if cmpTypes(c, lhs.typ, rhsTyp) in {isGeneric, isEqual}:
             c.config.internalAssert c.p.resultSym != nil
             # Make sure the type is valid for the result variable
-            typeAllowedCheck(c, n.info, rhsTyp, skResult)
-            lhs.typ = rhsTyp
+            let typ = typeAllowedOrError(rhsTyp, skResult, c, rhs, {})
+            if typ.isError:
+              result[1] = typ.n
+              hasError = true
+
             c.p.resultSym.typ = rhsTyp
             c.p.owner.typ[0] = rhsTyp
           else:
-            # XXX: if this is an nkError, should we modify the rhs and the
-            #      overall assignment and return that, cascading upward?
-            let r = typeMismatch(c.config, n.info, lhs.typ, rhsTyp, rhs)
-            if r.kind == nkError:
-              result = n
-              result[1] = r # rhs error
-              result = c.config.wrapError(result)
-              return
-
-      n[1] = fitNode(c, le, rhs, goodLineInfo(n[1]))
-      when false: liftTypeBoundOps(c, lhs.typ, lhs.info)
-
-      result = fixAbstractType(c, n)
-      asgnToResultVar(c, result, n[0], n[1])
-  result = n
+            result[1] = typeMismatch(c.config, n.info, lhs.typ, rhsTyp, rhs)
+            hasError = true
+
+      if not hasError:
```

Comment:
The old code and diff are a bit tricky to follow, but I _believe_ this has to guard on `if result[1].kind != nkError`, because otherwise we cut the rhs analysis short and that _might_ be why CI is failing. Just a guess.